lib/ui/trash_page.dart

```diff
@@ -1,30 +1,57 @@
+import 'dart:ui';
+
 import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
 import 'package:photos/core/event_bus.dart';
 import 'package:photos/db/trash_db.dart';
+import 'package:photos/ente_theme_data.dart';
 import 'package:photos/events/files_updated_event.dart';
 import 'package:photos/events/force_reload_trash_page_event.dart';
 import 'package:photos/models/galleryType.dart';
 import 'package:photos/models/selected_files.dart';
+import 'package:photos/ui/common/bottomShadow.dart';
 import 'package:photos/ui/gallery.dart';
 import 'package:photos/ui/gallery_app_bar_widget.dart';
 import 'package:photos/ui/gallery_overlay_widget.dart';
+import 'package:photos/utils/delete_file_util.dart';
 
-class TrashPage extends StatelessWidget {
+class TrashPage extends StatefulWidget {
   final String tagPrefix;
   final GalleryType appBarType;
   final GalleryType overlayType;
   final _selectedFiles = SelectedFiles();
+  TrashPage(
+      {this.tagPrefix = "trash_page",
+      this.appBarType = GalleryType.trash,
+      this.overlayType = GalleryType.trash,
+      Key key})
+      : super(key: key);
 
-  TrashPage({
-    this.tagPrefix = "trash_page",
-    this.appBarType = GalleryType.trash,
-    this.overlayType = GalleryType.trash,
-    Key key,
-  }) : super(key: key);
+  @override
+  State<TrashPage> createState() => _TrashPageState();
+}
+
+class _TrashPageState extends State<TrashPage> {
+  Function() _selectedFilesListener;
+  @override
+  void initState() {
+    _selectedFilesListener = () {
+      setState(() {});
+    };
+    widget._selectedFiles.addListener(_selectedFilesListener);
```

Comment:
Selecting any files in the trash page will refresh the entire page. Instead of that, can we create a separate widget which will be either empty or visible based on selected files?